Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -941,6 +941,8 @@ std::unique_ptr<RenderStyle> TreeResolver::resolveAgainInDifferentContext(const
     Adjuster adjuster(m_document, parentStyle, resolutionContext.parentBoxStyle, !styleable.pseudoElementIdentifier ? &styleable.element : nullptr);
     adjuster.adjust(*newStyle);
 
+    newStyle->copyPseudoElementBitsFrom(*resolvedStyle.style);
```

Comment:
Out of curiosity: why do these bits not include the pseudo element type?

Reply:
They tell what pseudo-elements the associated element has. `RenderStyle::pseudoElementType()` tells which pseudo-element the current style is for.